shared/src/main/scala/com/raquo/waypoint/Route.scala

```diff
@@ -393,8 +475,39 @@ object Route {
     )
   }
 
-  private def matchPageByClassTag[Page: ClassTag, Args](encode: Page => Args): PartialFunction[Any, Args] = {
-    case page: Page => encode(page)
+  /** Create a route for a static page that does not encode any data in the URL.
+    *
+    * Note that this is only a total route when the [[Page]] type refers to an object type. For example, this is OK:
+    * {{{
+    * object Foo extends AppPage
+    *
+    * val route: Total[Foo.type, Unit] = Route.staticTotal(Foo, root / "foo" / endOfSegments)
+    * }}}
+    *
+    * However, it is possible to create a route which is not actually total:
+    * {{{
+    * object Foo extends AppPage
+    *
+    * val route: Total[AppPage, Unit] = Route.staticTotal[AppPage](Foo, root / "foo" / endOfSegments)
+    * }}}
+    *
+    * This will think that it can route any `AppPage`, however it will not. Use [[static]] for those cases instead.
+    * */
+  def staticTotal[Page](
+    staticPage: Page,
+    pattern: PathSegment[Unit, DummyError],
+    basePath: String = ""
+  ): Total[Page, Unit] = {
```

Comment:
So, manually specifying the `AppPage` in `Route.staticTotal[AppPage](...)`  is required for the bug to manifest.

It's unfortunate, but that alone is not the entire problem. The following would also produce non-total Total routes:

 ```scala
case class LoginPage(spte: Option[String])

Route.staticTotal(LoginPage(step = Some("blah")), root / "login" / blah /endOfSegments)

Route.staticTotal(LoginPage(step = None), root / "login" / endOfSegments)
```

Scala would infer the total type to be `LoginPage`, but that would not be true for either of these routes, in fact there is no type that could make it a total type.

So, static pages can only be a Total route if they refer to singleton objects, AND the user did not put an incorrect type ascription. Which is the typical use case, but still, this feels rather fragile.

And yet, again, I don't see a better solution. Unless, perhaps Scala 3 macros can be used to check that the provided `Page` type is a singleton type? I dunno. I would be ok with that method being Scala 3 only.

As it stands with this PR, I think, most users will keep using the `static` method, but to add some extra warning, let's rename `staticTotal` to `unsafeStaticTotal`.

Also, can you please add a small test to show that `staticTotal` works as expected, and to see how exactly your a non-total version of it fails – if it's some inscrutable error, we should provide a nicer exception.

With the rename and the tests, I think we'll be able to merge it. Thanks for your work, and sorry for the delays, been very busy with another project.

Reply:
> but to add some extra warning, let's rename staticTotal to unsafeStaticTotal.

OK, but I'll name it `staticTotalUnsafe`, just because Metals autocomplete doesn't search in the middle of the word, so `unsafeStaticTotal` wouldn't show up in autocomplete for `.static`, whilst `.staticTotalUnsafe` would.

Reply:
Sure ok